agents_infra/agents_infra/utils/file_utils.py

```diff
@@ -0,0 +1,81 @@
+import base64
+import codecs
+import json
+import os
+import re
+import time
+
+
+def ensure_directory_exists(path):
+    """Ensure the parent directory for the given file path exists."""
+    try:
+        dir_path = os.path.dirname(path)
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+    except Exception as e:
+        raise IOError('Failed to create directory for path %s: %s' % (path, e))
+
+
+def write_to_file(path, content, logger=None):
+    """Write content to a file, creating parent directories as needed."""
+    try:
+        ensure_directory_exists(path)
+        with codecs.open(path, 'w', encoding='utf-8') as f:
+            f.write(content)
+    except Exception as e:
+        if logger:
+            logger.error('Failed to write to file %s: %s' % (path, e))
+        raise IOError('Failed to write to file %s: %s' % (path, e))
+
+
+def read_from_file(path, logger=None):
+    """Read and return the content of a file."""
+    try:
+        if os.path.exists(path):
+            with codecs.open(path, 'r', encoding='utf-8') as f:
+                return f.read()
+    except Exception as e:
+        if logger:
+            logger.error('Failed to read from file %s: %s' % (path, e))
+        raise IOError('Failed to read from file %s: %s' % (path, e))
+    return None
+
+
+def is_jwt(token):
+    """Basic pattern check to see if the string looks like a JWT."""
+    try:
+        jwt_pattern = re.compile(
+            r'^[A-Za-z0-9\-_]+\.[A-Za-z0-9\-_]+\.[A-Za-z0-9\-_]+$'
+        )
+        return bool(jwt_pattern.match(token))
+    except Exception:
+        return False
+
+
+def decode_jwt(token):
+    """Decode the payload part of a JWT (without verifying signature)."""
+    try:
+        parts = token.split('.')
+        if len(parts) != 3:
+            return {}
+        payload = parts[1]
+        padding = '=' * (4 - len(payload) % 4)
+        payload += padding
```

Comment:
This can add too much padding if 'len(payload) % 4 == 0'. Safer: 'payload += '=' * (-len(payload) % 4)'.
Use 'base64.urlsafe_b64decode' for JWTs, not 'base64.b64decode'.